src/iceberg/test/rest_catalog_test.cc

```diff
@@ -337,4 +410,78 @@ TEST_F(RestCatalogIntegrationTest, DropNamespace) {
   EXPECT_FALSE(*exists_result);
 }
 
+TEST_F(RestCatalogIntegrationTest, CreateTable) {
+  auto catalog_result = CreateCatalog();
+  ASSERT_THAT(catalog_result, IsOk());
+  auto& catalog = catalog_result.value();
+
+  // Create nested namespace with properties
+  Namespace ns{.levels = {"test_create_table", "apple", "ios"}};
+  std::unordered_map<std::string, std::string> ns_properties{{"owner", "ray"},
+                                                             {"community", "apache"}};
+
+  // Create parent namespaces first
+  auto status = catalog->CreateNamespace(Namespace{.levels = {"test_create_table"}}, {});
+  EXPECT_THAT(status, IsOk());
+  status =
+      catalog->CreateNamespace(Namespace{.levels = {"test_create_table", "apple"}}, {});
+  EXPECT_THAT(status, IsOk());
+  status = catalog->CreateNamespace(ns, ns_properties);
+  EXPECT_THAT(status, IsOk());
+
+  // Create schema
+  auto schema = std::make_shared<Schema>(
+      std::vector<SchemaField>{SchemaField::MakeOptional(1, "foo", string()),
+                               SchemaField::MakeRequired(2, "bar", int32()),
+                               SchemaField::MakeOptional(3, "baz", boolean())},
+      /*schema_id=*/1);
+
+  // Create partition spec and sort order (unpartitioned and unsorted)
+  auto partition_spec_result = PartitionSpec::Make(PartitionSpec::kInitialSpecId, {}, 0);
+  ASSERT_THAT(partition_spec_result, IsOk());
+
+  auto sort_order_result =
+      SortOrder::Make(SortOrder::kUnsortedOrderId, std::vector<SortField>{});
+  ASSERT_THAT(sort_order_result, IsOk());
+
+  // Create table
+  TableIdentifier table_id{.ns = ns, .name = "t1"};
+  std::unordered_map<std::string, std::string> table_properties;
+  auto table_result = catalog->CreateTable(table_id, *schema, **partition_spec_result,
+                                           **sort_order_result, "", table_properties);
+  ASSERT_THAT(table_result, IsOk());
+  auto& table = table_result.value();
+
+  // Verify table
+  EXPECT_EQ(table->name().ns.levels,
+            (std::vector<std::string>{"test_create_table", "apple", "ios"}));
+  EXPECT_EQ(table->name().name, "t1");
+
+  auto table_schema_result = table->schema();
```

Comment:
Can we simplify the verification below? It looks too lengthy. Perhaps we can just verify the table name is good.